docs/notebooks/montecarlo.pct.py

```diff
@@ -0,0 +1,210 @@
+#
+# The optimization procedure will benefit from having some starting data from the objective function to base its search on. We sample five points from the search space and evaluate them on the observer. We can represent the search space using a `Box`.
+
+# %%
+observer = mk_observer(branin, OBJECTIVE)
+lower_bound = tf.cast(mins, gpflow.default_float())
+upper_bound = tf.cast(maxs, gpflow.default_float())
+search_space = trieste.space.Box(lower_bound, upper_bound)
+
+num_initial_points = 10
+initial_query_points = search_space.sample(num_initial_points)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# ## Model the objective function
+#
+# The Bayesian optimization procedure estimates the next best points to query by using a probabilistic model of the objective. We'll use Gaussian process regression for this, provided by GPflow. The model will need to be trained on each step as more points are evaluated, so we'll package it with GPflow's Scipy optimizer.
+#
+# Note we could leave it to the optimizer to build this model, but we'll want to inspect it later so we'll build it ourselves.
+#
+# Just like the data output by the observer, the optimization process assumes multiple models, so we'll need to label the model in the same way.
+
+# %%
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=0.2 * np.ones(2,))
+    gpr = gpflow.models.GPR(astuple(data), kernel, noise_variance=1e-5)
+    set_trainable(gpr.likelihood, False)
+
+    return {OBJECTIVE: trieste.models.create_model({
+        "model": gpr,
+        "optimizer": gpflow.optimizers.Scipy(),
+        "optimizer_args": {"options": dict(maxiter=100)},
+    })}
+
+model = build_model(initial_data[OBJECTIVE])
+
+# %% [markdown]
+# # Try an alternative acquisition function
+
+# %% [markdown]
+# By default, Trieste uses Expected Improvement (EI) as its acquisition function when performing Bayesian optimization. However, many alternative acqusition functions have been developed. One such alternative is Max-value Entropy Search (MES), which approximates the distribution of current estimate of the global minimum and tries to decrease its entropy with each optimization step.
+
+# %% [markdown]
+# We plot these two acquisition functions across our search space. Areas with high acquisition function scores (i.e bright regions) are those rated as promising locations for the next evaluation of our objective function. We see that EI wishes to continue exploring the search space, whereas MES wants to focus resources on evaluating a specific region.
```

Comment:
Change text. The important points are : 
- both surfaces look similar
- the MC-EI is smooth because we use the reparametrization trick

Reply:
we're not including the notebooks in this PR are we? It will take longer if we do

Reply:
also, this paragraph is misplaced, it should appear later. We need a paragraph here that explains that we create a MC-EI acquisition for the sake of demonstrating how to do it